### PR TITLE
DG update - Anli

### PR DESCRIPTION
**Pull request by ZhangAnli**
Added Use Cases. 
Added NFRs. 
Added User Stories.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -198,6 +198,12 @@ that uses the list of keywords to find the search results based on the supplied
 This predicate is passed into the `ModelManager`'s `updateFilteredEntityList()` method and subsequently generates the
 CommandResult instance that is then passed on in the LogicManager.
 
+Below is an example sequence diagram for a valid view command from the user.
+
+![FindActivityDiagram](images/FindSequenceDiagram.png)
```
Comment (CharlesLee01) [inline docs/DeveloperGuide.md:203]: Should be [FindSequenceDiagram]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -211,8 +217,11 @@ Based on the class type of the target entity, we will reveal the search results
 
 This list is subsequently passed on to the `RelatedEntityPredicate` that will later be used in the ModelManager's `updatefilteredEntityList())` method to finally reveal the search results.
 
+Below is an example sequence diagram for a valid view command from the user. 
 
-Below is an example activitiy diagram for a valid view command from the user.
+![ViewActivityDiagram](images/ViewSequenceDiagram.png)
```
Comment (CharlesLee01) [inline docs/DeveloperGuide.md:222]: Should be [ViewSequenceDiagram]

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -380,69 +468,87 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 **Extensions**
 
-- 2a. The given command/format is invalid.
-    - 2a1. Pawbook shows an error message to the user.
-    Use case resumes at step 2.
+- 1a. The given command/format is invalid. 
+    - 1a1. Pawbook shows an error message to the user.
+    - 1a2. User supplies the correct command. <br>
+      Steps 1a1-1a2 are repeated until the command entered is correct.
+    Use case resumes at step 3.
```
Comment (branzuelajohn) [inline docs/DeveloperGuide.md:510]: Shouldnt this be step 2

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -380,69 +468,87 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 6.  A simple interface that is easy to navigate.
 7.  Not required to handle finance-related bookkeeping.
-8.  Should not crash or throw unexpected errors when internet connection is
-    unavailable.
+8.  Pawbook does not require internet connection to run. 
+9.  Able to work in all different time zones, regardless of where the system is operated in the world. 
+10. Features should be implemented such that they can undergo automated testing. 
+11. Pawbook data should be saved locally. Not possible to be hacked from external machines. Only way to access is through the local machine.
+12. Only one Pawbook program can run on a local machine at any time. 
+13. Pawbook program file size should not exit beyond 1GB.
```
Comment (branzuelajohn) [inline docs/DeveloperGuide.md:550]: exceed?

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (wei-yutong) [reply docs/DeveloperGuide.md:510]: good catch!

Comment (branzuelajohn) [reply docs/DeveloperGuide.md:510]: Thanks!

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -198,6 +198,12 @@ that uses the list of keywords to find the search results based on the supplied
 This predicate is passed into the `ModelManager`'s `updateFilteredEntityList()` method and subsequently generates the
 CommandResult instance that is then passed on in the LogicManager.
 
+Below is an example sequence diagram for a valid view command from the user.
+
+![FindActivityDiagram](images/FindSequenceDiagram.png)
+
+Below is an example activity diagram for a valid view command from the user.
```
Comment (shaelynl) [inline docs/DeveloperGuide.md:205]: should be valid find command?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -282,18 +291,22 @@ However, this requires there to be no duplicate dog or program names.
...
+| -------- | -------------------- | -------------------------------------------------------------------------- | ---------------------------------------------------------------------- |
+| `* * *`  | Dog school manager   | Have a list of dogs, owners, and programs in the school                    | Keep track of the operations and dogs we are responsible for     |
+| `* * *`  | Dog school manager   | Easily switch between the different lists for dogs, owners and programs    | Quickly view all the profiles of one entity type     |
+| `* * *`  | Dog school manager   | Add dog/owner/program profiles                                             | Keep track of the operations and parties involved in the school|
+| `* * *`  | Dog school manager   | Delete dog/owner/program profiles                                          | Keep track of the operations and parties involved in the school|
+| `* * *`  | Dog school manager   | Edit a dog/owner/program profile                                           | Correct/update a profile when it is incorrect.|
```
Comment (wei-yutong) [inline docs/DeveloperGuide.md:300]: you ended the sentence with a fullstop here but not the other user stories. should choose either all fullstops or none for consistency

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -282,18 +291,22 @@ However, this requires there to be no duplicate dog or program names.
...
+| `* * *`  | Dog school manager   | View a dog/owner/program profile                                           | Easily find out information on the target party|
+| `* * *`  | Dog school manager   | Find profiles using keywords instead of ID                                 | Easily find a target dog/owner/program very quickly |
+| `* * *`  | Dog school manager   | Enrol dogs into a specific dog program                                     | Dogs who recently joined a program are added to the class list |
+| `* *`    | Dog school manager   | Drop dogs out of a specific dog program                                    | Dogs that have left the class are no longer in the class list  |
+| `* * *`  | Dog school manager   | See the schedule for any day                                               | Easily view my schedule to know what programs are happening on that day. |
+| `* *`    | Dog school manager   | Autosave the data after every command                                      | Regularly save the data and protect sensitive data in the event that the system crashes.  |
```
Comment (wei-yutong) [inline docs/DeveloperGuide.md:306]: here as well

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -380,69 +468,87 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 **Extensions**
 
-- 2a. The given command/format is invalid.
-    - 2a1. Pawbook shows an error message to the user.
-      Use case resumes at step 2.
+- 1a. The given command/format is invalid.
+    - 1a1. Pawbook shows an error message to the user.
+    - 1a2. User supplies the correct command. <br>
+      Steps 1a1-1a2 are repeated until the command entered is correct.
+      Use case resumes at step 3.
```
Comment (wei-yutong) [inline docs/DeveloperGuide.md:530]: should be step 2?

**Pull request closed without merging**